src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -957,7 +957,27 @@ protected bool QualifiedIdentifier(out int identifierLength)
 
         protected bool NamespaceOrTypeName()
         {
-            if (Optional(CSharpSymbolType.Identifier) || Optional(CSharpSymbolType.Keyword))
+            if (Optional(CSharpSymbolType.LeftParenthesis))
+            {
+                while (!Optional(CSharpSymbolType.RightParenthesis) && !EndOfFile)
+                {
```

Comment:
`(   string val, int val2)` is a valid tuple. Therefore, need `Optional(CSharpSymbolType.WhiteSpace)` here (means you can 
 also remove the trailing one at the end of this while.